### PR TITLE
Fix call to `extractall`

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* There was yet one call that applied to tar archives as well.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/shell/sagemaker/dyn.py [hunk 1]
```diff
@@ -78,10 +84,10 @@ def install_requirement(self, path: Path):
     def install(self, path):
         if path.is_file():
             if tarfile.is_tarfile(path):
-                self.handle_archive(tarfile.open, path)
+                self.handle_archive(tarfile.open, safe_extractall, path)
 
             elif zipfile.is_zipfile(path):
-                self.handle_archive(zipfile.ZipFile, path)
+                self.handle_archive(zipfile.ZipFile, extractall, path)
```
Comment (jaheba) [inline src/gluonts/shell/sagemaker/dyn.py]:
```suggestion
                self.handle_archive(zipfile.ZipFile, zipfile.ZipFile.extractall, path)
```

This should also work.

**Comment by jaheba** [comment]
Let's check when we added `dyn`, I don't think we need to backport this that far back.

**Comment by lostella** [comment]
was there in 0.9.x https://github.com/awslabs/gluonts/blob/v0.9.x/src/gluonts/shell/sagemaker/dyn.py

**Comment by jaheba** [comment]
I'm getting old!

**Review by jaheba — APPROVED** [review]

**Pull request merged**